File: BOJ/etc/2606.py
"""
- BOJ 2606
- CDR
"""
from sys import stdin
n = int(stdin.readline())
m = int(stdin.readline())
graph = [[] * n for _ in range(n + 1)]
for _ in range(m):  # graph 만들기
    a, b = map(int, stdin.readline().split())
    # ex. 1 - 2 는 node 1 과 node 2 에 모두 엮여있는 것으로 graph 에 둘다 넣어줘야 한다
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (n + 1)
count = 0


def dfs(graph, start, visited):
    global count
    visited[start] = True
    for i in graph[start]:
        if not visited[i]:
            dfs(graph, i, visited)
            count += 1


dfs(graph, 1, visited)
print(count)

File: BOJ/etc/2606.js
const readline = require('readline');
const r1 = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

let input = []

r1.on("line", (data) => {
    input.push(data.toString())
}).on("close", () => {
    let n = parseInt(input.shift())
    let m = parseInt(input.shift())
    let graph = Array.from(Array(n + 1), () => Array())
    let visited = Array.from(Array(n + 1).fill(false))
    let cnt = 0;

    for (let i of input) {
        let [x, y] = i.split(' ').map(Number)
        graph[x].push(y)
        graph[y].push(x)
    }

    const bfs = (start) => {
        visited[start] = true
        let q = [start]
        while (q.length > 0) {
            let num = q.shift()
            for (let i of graph[num]) {
                if (visited[i] === false) {
                    visited[i] = true
                    q.push(i)
                    cnt += 1
                }
            }
        }
    }
    bfs(1)

    console.log(cnt)
})